### PR TITLE
Add SOCKS proxy support

### PR DESCRIPTION
**Pull request by vincenzo**
## Description

Adds the ability to tell the CLI to use a SOCKS proxy when connecting to the API. 

The CLI will look for a `VIP_PROXY` environment variable. If found, it will use its value to add the proxy on the connection. 

## Steps to Test

TBD

### CODE REVIEW

File: src/lib/api.js [hunk 1]
```diff
@@ -49,7 +50,9 @@ export default async function API(): Promise<ApolloClient> {
 		}
 	} );
 
-	const httpLink = new HttpLink( { uri: API_URL, headers: headers } );
+	const httpLink = new HttpLink( { uri: API_URL, headers: headers, fetchOptions: {
+		agent: process.env.hasOwnProperty( 'VIP_PROXY' ) ? ProxyAgent( process.env.VIP_PROXY ) : null,
```
Comment (nickdaugherty) [inline src/lib/api.js]: Should this be `new ProxyAgent()`?
```suggestion
		agent: process.env.hasOwnProperty( 'VIP_PROXY' ) ? new ProxyAgent( process.env.VIP_PROXY ) : null,
```

Comment (vincenzo) [reply src/lib/api.js]: It should indeed.

**Review by nickdaugherty — APPROVED** [review]

**Review by chrean — APPROVED** [review]
👍

**Comment by chrean** [comment]
Once this is shipped and its release notes published, we can proceed merging and deploying the other part.

**Comment by vincenzo** [comment]
Proxied connection verified.

**Comment by nickdaugherty** [comment]
Woohoo!

**Pull request merged**